src/connector/src/sink/big_query.rs

```diff
@@ -284,16 +364,43 @@ impl Sink for BigQuerySink {
             .common
             .build_client(&self.config.aws_auth_props)
             .await?;
+        let BigQueryCommon {
+            project: project_id,
+            dataset: dataset_id,
+            table: table_id,
+            ..
+        } = &self.config.common;
+
+        if self.config.common.auto_create_table {
+            match client
+                .table()
+                .get(project_id, dataset_id, table_id, None)
+                .await
+            {
+                Err(BQError::ResponseError { error: _ }) => {
+                    _ = self.create_table(
+                        &client,
+                        project_id,
+                        dataset_id,
+                        table_id,
+                        &self.schema.fields,
+                    );
+                }
+                Err(e) => return Err(SinkError::BigQuery(e.into())),
+                _ => {}
+            }
+        }
+
         let mut rs = client
```

Comment:
When auto_create_table=true, it seems that there is no need to query the schema again